cpp/src/rolling/detail/rolling_utils.cuh

```diff
@@ -17,14 +17,74 @@
 #pragma once
 
 #include <cudf/detail/iterator.cuh>
+#include <cudf/detail/rolling.hpp>
 #include <cudf/types.hpp>
 
 #include <cuda/functional>
+#include <cuda/std/type_traits>
 
 namespace CUDF_EXPORT cudf {
 
 namespace detail::rolling {
 
+/**
+ * @brief Information about group bounds of the current row's group.
+ */
+struct range_group_info {
```

Comment:
I'm not sure that moving all the methods here was an improvement, but at this point I think that's probably just an indication of necessary complexity just getting shoved around so no need to iterate further. The current version is plenty good.